src/components/notifications/AppExplorer/AppCard/index.tsx

```diff
@@ -64,26 +67,23 @@ const AppCard: React.FC<AppCardProps> = ({
 
       setSubscribing(true)
       try {
-        await notifyClientProxy?.subscribe({
-          account: userPubkey,
-          appDomain: new URL(url).host
-        })
+        await subscribe(undefined, host)
       } catch (error) {
         logError(error)
         setSubscribing(false)
         showErrorMessageToast(`Failed to subscribe to ${name}`)
       }
     },
-    [userPubkey, name, description, logo, url, setSubscribing, subscribed, notifyClientProxy]
+    [userPubkey, name, description, logo, url, setSubscribing, subscribed]
```

Comment:
Another thing to get rid of; `useMemo`s and `useCallback`s. Unless we have big calculations and expensive logics, we should avoid using these since the following their dependencies can be big headaches for us in the future. 

In our case, the `handleSubscription` function gets almost all the states/properties of the component which makes the memoization useless. So eventually the component will be rerendered once one of these dependencies changes. For me, we are making it harder to maintain our code.

Also, AFAIK, handling the memoization by default is one of the plans of the React team. So unless we have performance issues (with extra rerenders), I think we should avoid memoization.

Reply:
As discussed, let's keep those for separate PR